### DISCUSSION TITLE
Apparently the pico can handle 16 PWM ch, is this do-able?

### DISCUSSION THREAD

**Discussion by tiro-uspsss** [tjko/fanpico#37] [Q&A]
1604 model anyone?

I'd be keen for a 16 channel model.. wouldn't need the 4 PWM inputs either actually

cheers

**Comment by tjko** [comment]
"1604" is not easily done, but "1600" should be relatively easy to implement (16 independent fan outputs, but no "motherboard" inputs).  With "1600" probably would need to multiplex temperature sensors, but that in turn would allow havig up to 8 temperature sensors.

**Comment by tiro-uspsss** [comment]
I'd be fine with that.. 

somewhat of a topic change... is it possible to feed fanpico temp data from sensors in the rest of the system (say hdd, cpu, gpu etc), & have a preset PWM curve to respond to temp changes?

I live in a rather cool/cold area, & don't want my HDDs getting too cold (I want them to run between 25-40C). I would love for fanpico to control the fans such that the HDDs don't go out of this range... so in winter, the fans most likely would be off. In summer, probably only a ~50% load.

**Reply by tjko** [reply]
Current firmware supports direct control of fans by a script/program running on a host computer by specifying "fixed" PWM duty cycle (see https://github.com/tjko/fanpico/blob/main/commands.md#configurefanxsource)

I've been thinking of adding support for "registers" that would store temperature values, that external program would update.
Then these "registers" could be configured as "source" for fan speed control (just like PWM input from motherboard or the temperature sensors currently).
